### PR TITLE
Updated versions: gradle, java, jacoco, kotlin, etc

### PR DESCRIPTION
**Pull request by leventeBajczi**
This PR updates the versions of Java and Gradle, as well as some dependencies.

### CODE REVIEW

**Review by as3810t — APPROVED** [review]
Looks alright to me. Although this PR does not update the Gradle version Theta uses. Is the Gradle version up-to-date enough?

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.6.1-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
```
Comment (leventeBajczi) [inline gradle/wrapper/gradle-wrapper.properties:3]: @as3810t I think this should update Gradle to 7.4.2. See build logs:
```
$ ./gradlew -v

------------------------------------------------------------
Gradle 7.4.2
------------------------------------------------------------

```

Comment (as3810t) [reply gradle/wrapper/gradle-wrapper.properties:3]: You are right, my mistake!

Comment (leventeBajczi) [reply gradle/wrapper/gradle-wrapper.properties:3]: Btw I used `./gradlew wrapper --gradle-version 7.4.2` to achieve this

**Pull request merged**